### ISSUE TITLE
warnOnceIfEsmOnlyPackage erroneously warned for marked

### ISSUE THREAD

**Issue by jamesarosen** [remix-run/remix#4324]
### What version of Remix are you using?

1.6.5

### Steps to Reproduce

Add `marked` to dependencies, run `remix run`.

### Expected Behavior

No warning message, since [marked](https://www.npmjs.com/package/marked) publishes CJS and ESM in the same package:

```json
"main": "./lib/marked.cjs",
"module": "./lib/marked.esm.js",
"browser": "./lib/marked.umd.js",
```

https://github.com/markedjs/marked/pull/2281 has some relevant history.

### Actual Behavior

`remix dev` emits

> marked is possibly an ESM only package and should be bundled with "serverDependenciesToBundle in remix.config.js.

**Comment by machour** [comment]
@jamesarosen could you try with the latest Remix version?

**Comment by pcattori** [comment]
Fixed by #6916
